bindings/cs/MongoDB.Crypt/IKmsKeyId.cs

```diff
@@ -37,5 +48,31 @@ public interface IKmsKeyId
     internal interface IInternalKmsKeyId
     {
         void SetCredentials(ContextSafeHandle handle, Status status);
+        void SetAlternateKeyNames(ContextSafeHandle handle, Status status);
+    }
+
+    /// <summary>
+    /// Using extension methods in lieu of default interface methods that will not be available until C#8 releases
+    /// </summary>
+    internal static class KmsExtensions
+    {
+        internal static void SetAlternateKeyNames(this IKmsKeyId kmsKeyId, ContextSafeHandle handle, Status status)
```

Comment:
Where are the alternate key names? Are they in the handle? If so we might want to rename.

Reply:
I don't think this is an appropriate extension method on IKmsKeyId.

The operation is not being performed ON the IKmsKeyId. It is being applied ON the context right?

Probably shouldn't be an extension method at all. We should find an appropriate home for this method.

Reply:
Let's Zoom about this tomorrow!

Reply:
Agreed. Go ahead and schedule something that works for you.

Reply:
Here's the problem. Consider this code:

    person.SetName(name);

It is clear that we are setting the Name property of a Person object to a new value.

Now consider this code:

    keyId.SetAlternateKeyNames(context, handle);

One would assume by common object oriented naming conventions that this code is setting the AlternateKeyNames property of a KeyId object to a new value.

But it's NOT! Apparently it is *taking* the existing AlternateKeyNames from the keyId object and propagating them to the context.

This would make more sense if it was written:

    context.SetAlternateKeyNames(keyId, handle);

So the object being affected was the receiver of the method call.

However... given time constraints the consensus seems to simply ignore the confusion caused by this implementation and move on.